Ingress/controllers/gce/backends.go

```diff
@@ -40,7 +40,17 @@ func portKey(port int64) string {
 }
 
 func beName(port int64) string {
-	return fmt.Sprintf("%v-%d", backendPrefix, port)
+	// TODO: Pipe the clusterName through, for now it saves code churn to just
+	// grab it globally, especially since we haven't decided how to handle
+	// namespace conflicts in the Ubernetes context.
+	if *clusterName == "" {
+		return fmt.Sprintf("%v-%d", backendPrefix, port)
+	}
+	taggedName := fmt.Sprintf("%v-%d%v%v", backendPrefix, port, clusterNameDelimiter, *clusterName)
+	if len(taggedName) > nameLenLimit {
+		taggedName = taggedName[:nameLenLimit]
+	}
+	return taggedName
```

Comment:
I actually just pass the test namespace so it should be unique eitherway, and contains a similar hash. I'd assumed if users specify colliding cluster names that's their problem? Not against your hash idea though, can I do it in a follow-up?